3761/1/gem_statisic_fetcher.rb

```diff
@@ -0,0 +1,43 @@
+require './page.rb'
+class GemStatisicFetcher
+  attr_reader :gem_name, :agent
+  attr_reader :stats
+
+  def self.fetch_statistic(gem_name, agent)
+    fetcher = new(gem_name, agent)
+    fetcher.fetch_statistic
+    fetcher.analise_statistic
+    fetcher.stats
+  end
+
+  def initialize(gem_name, agent)
+    @gem_name = gem_name
+    @agent = agent
+    @stats = {}
+  end
+  # :reek:TooManyStatements
+  # rubocop:disable Metrics/AbcSize
+
+  def fetch_statistic
+    page_gem_in_ribygems = Page.new('https://rubygems.org/gems/' + gem_name, agent)
+    return if page_gem_in_ribygems.data == false
+    page_gem_in_github = Page.new(page_gem_in_ribygems.url_in_github, agent)
+    url_with_used = page_gem_in_ribygems.url_in_github + '/network/dependents'
+    page_gem_with_used_by = Page.new(url_with_used, agent)
+    @stats = page_gem_in_github.find_data
+    @stats[:used_by] = page_gem_with_used_by.find_used_by
+    @stats[:gem] = gem_name
+  end
+  # rubocop:enable Metrics/AbcSize
+
+  RATIO = { used_by: 0.9, watch: 0.8, star: 1, fork: 0.85,
+            contributors: 0.75, issues: 0.75 }.freeze
+
+  def analise_statistic
```

Comment:
Why you make calculation in Fetcher class? Read about single responsibility principle